packages/mongodb-chatbot-eval/src/Pipeline.ts

```diff
@@ -0,0 +1,135 @@
+import { ObjectId, logger } from "mongodb-rag-core";
+import { CommandRunMetadata } from "./CommandMetadataStore";
+import { ConfigConstructor, EvalConfig } from "./EvalConfig";
+import { strict as assert } from "assert";
+import { generateDataAndMetadata } from "./generate";
+import { generateEvalsAndMetadata } from "./evaluate/generateEvalsAndMetadata";
+import { generateReportAndMetadata } from "./report";
+
+type PipelineGenerateFunc = (name: string) => Promise<CommandRunMetadata>;
+
+type PipelineEvaluateFunc = (
+  name: string,
+  generatedDataRunId: ObjectId
+) => Promise<CommandRunMetadata>;
+
+type PipelineReportFunc = (
+  name: string,
+  evalResultsRunId: ObjectId
+) => Promise<CommandRunMetadata>;
+
+export type PipelineFunc = (
```

Comment:
```suggestion
export type Pipeline = (
```

I care a bit less about `FooFunc` in type names though personally I don't think it's necessary.